src/validators/is_instance.rs

```diff
@@ -56,10 +55,14 @@ impl Validator for IsInstanceValidator {
         _state: &mut ValidationState<'_, 'py>,
     ) -> ValResult<PyObject> {
         let Some(obj) = input.as_python() else {
-            return Err(ValError::InternalErr(PyNotImplementedError::new_err(
-                "Cannot check isinstance when validating from json, \
-                            use a JsonOrPython validator instead.",
-            )));
+            let method_name = "isinstance".to_string();
+            return Err(ValError::new(
+                ErrorType::CannotValidateFromJson {
+                    context: None,
+                    method_name,
+                },
+                input,
+            ));
         };
```

Comment:
In the 0.01% chance we eventually support another text format, I guess that these validators won't accept that format either. Should the error be turned around? e.g. `needs_python_object.

```suggestion
            let method_name = "isinstance".to_string();
            return Err(ValError::new(
                ErrorType::NeedsPythonObject {
                    context: None,
                    method_name,
                },
                input,
            ));
        };
```